Sources/Apodini/Semantic Model Builder/Model/Endpoint.swift

```diff
@@ -158,104 +164,50 @@ class EndpointsTreeNode {
-                                        + "You can't have multiple path parameters on the same level!")
-                        } else {
-                            childContainsPathParameter = true
-                        }
-                    }
-                    
-                    child = EndpointsTreeNode(path: first, parent: self)
-                    nodeChildren[first.description] = child
-                }
-                
-                // swiftlint:disable:next force_unwrapping
-                child!.addEndpoint(&endpoint, at: pathComponents)
-            } else {
-                fatalError("Encountered PathComponent which isn't a _PathComponent!")
-            }
-        }
-    }
-    
-    private func collectAbsolutePath(_ absolutePath: inout [_PathComponent]) {
-        guard let parent = parent else {
-            return
-        }
-        
-        parent.collectAbsolutePath(&absolutePath)
-        absolutePath.append(path)
-    }
-    
-    func relativePath(to node: EndpointsTreeNode) -> [_PathComponent] {
-        var relativePath: [_PathComponent] = []
-        collectRelativePath(&relativePath, to: node)
-        return relativePath
-    }
-    
-    private func collectRelativePath(_ relativePath: inout [_PathComponent], to node: EndpointsTreeNode) {
-        if node === self {
-            return
-        }
-        guard let parent = parent else {
-            return
-        }
-        
-        parent.collectRelativePath(&relativePath, to: node)
-        relativePath.append(path)
-    }
-    
-    fileprivate func collectRelationships(name: String, _ relationships: inout [EndpointRelationship]) {
-        if !endpoints.isEmpty {
-            relationships.append(EndpointRelationship(name: name, destinationPath: absolutePath))
             return
         }
         
-        for (childName, child) in nodeChildren {
-            // as Parameter is currently inserted into the path (which will change)
-            // checking against RequestInjectable is a lazy check to determine if this is a path parameter
-            // or just a regular path component. To be adapted.
-            let name = name + (child.path is RequestInjectable ? "" : "_" + childName)
-            child.collectRelationships(name: name, &relationships)
-        }
-    }
-    
-    /// This method prints the tree structure to stdout. Added for debugging purposes.
-    func printTree(indent: Int = 0) {
-        let indentString = String(repeating: "  ", count: indent)
-        
-        print(indentString + path.description + "/ {")
+        var pathComponents = paths
         
-        for (operation, endpoint) in endpoints {
-            print("\(indentString)  - \(operation): \(endpoint.description) [\(endpoint.identifier.rawValue)]")
+        guard let first = pathComponents.removeFirst() as? _PathComponent  else {
+            fatalError("Encountered PathComponent which isn't a _PathComponent!")
```

Comment:
I tried to think about covering this with a test case, my feeling is, that if possible at all, this would be a very artificial test case - how should this happen?